### PR TITLE
examples: update Android examples

### PR DESCRIPTION
**Pull request by ericgribkoff**
This addresses https://github.com/grpc/grpc-java/issues/3835 and https://github.com/grpc/grpc-java/issues/3938.

Code improvements include:

* Build file and dependency updates
* Bump targetSdkVersion
* Correcting the gradle wrapper for the clientcache example (correcting an earlier mistake on my part)
* Lint fixes (including making AsyncTask subclasses static)
* Dropping the m-prefix from member variables
* Fixing the code indentation
* Fixing and enabling proguard for the routeguide example.

The commits will be squashed on merged but can be reviewed separately. The first commit is the bulk of the updates, the second commit removes the m- prefix from the member fields, and the third was just running google-java-format to fix the indentation in routeguide and helloworld.

### CODE REVIEW

**Comment by zpencer** [comment]
The gradle-wrapper.jar still looks like it doesn't match up in the android examples, and across grpc-java:
```
$ find . -iname 'gradle-wrapper.properties' -exec md5sum {} \;
38624e4a98ce4c926170242993272188  ./examples/android/helloworld/gradle/wrapper/gradle-wrapper.properties
38624e4a98ce4c926170242993272188  ./examples/android/routeguide/gradle/wrapper/gradle-wrapper.properties
38624e4a98ce4c926170242993272188  ./examples/android/clientcache/gradle/wrapper/gradle-wrapper.properties
38624e4a98ce4c926170242993272188  ./examples/gradle/wrapper/gradle-wrapper.properties
38624e4a98ce4c926170242993272188  ./gradle/wrapper/gradle-wrapper.properties
$ find . -iname 'gradle-wrapper.jar' -exec md5sum {} \;
7d1da0885c00eeed3d7d104a648e0d5b  ./examples/android/helloworld/gradle/wrapper/gradle-wrapper.jar
8b4d70e732002adf1a6f54a2697e7a61  ./examples/android/routeguide/gradle/wrapper/gradle-wrapper.jar
8b4d70e732002adf1a6f54a2697e7a61  ./examples/android/clientcache/gradle/wrapper/gradle-wrapper.jar
a55b1dfcaaeb805d8337e5e3d0e0c5fc  ./examples/gradle/wrapper/gradle-wrapper.jar
956f845028b797bb67d1f490610ad496  ./gradle/wrapper/gradle-wrapper.jar
```

**Comment by ericgribkoff** [comment]
Hmm. It looks like https://github.com/grpc/grpc-java/commit/46b2ce19223f47367309e6fceda5517762156810#diff-3accdf551d92ee58ebb05f8763b47164 updated only the `gradle-wrapper.properties` files to upgrade to 4.3, while https://github.com/grpc/grpc-java/commit/0b50095aa62d2e593f03a929e5d53cbc4a30dc87#diff-3accdf551d92ee58ebb05f8763b47164 also updated the `gradle-wrapper.jar` files when upgrading to 4.0.1.

@ejona86 Since this is just the wrapper jar, do we actually care about ensuring these jars are all identical? It does seem strange we have several different checksums...

**Comment by ejona86** [comment]
The jars should generally match. It isn't essential to update a wrapper jar when bumping the gradle version (most of the time the only change in the JAR is a version number change). But if we do update it, we should be consistent. Let's not deal with a bug that only impacts one of the wrappers.

**Comment by ericgribkoff** [comment]
Left the jar as-is. Will unify them in separate PR (as it's not only different in the Android examples)

**Comment by ejona86** [comment]
> Will unify them in separate PR

SGTM

**Comment by ericgribkoff** [comment]
@zpencer friendly ping

File: examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/ClientCacheExampleActivity.java [hunk 1]
```diff
@@ -78,52 +80,62 @@ public void sendMessage(View view) {
...
-      mResultText.setText("");
+    private GrpcTask(Activity activity, SafeMethodCachingInterceptor.Cache cache) {
+      this.activityReference = new WeakReference<Activity>(activity);
+      this.cache = cache;
     }
 
     @Override
-    protected String doInBackground(Void... nothing) {
+    protected String doInBackground(Object... params) {
+      String host = (String) params[0];
```
[the remark this answers is not in the document]
Comment (ericgribkoff) [reply examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/ClientCacheExampleActivity.java]: This is the standard usage of AsyncTask - while we aren't actually executing them more than once here, the class is designed to support running multiple times with different arguments, so I'd prefer to keep these as-is.

[the remark this answers is not in the document]
Comment (zpencer) [reply examples/android/clientcache/app/src/main/java/io/grpc/clientcacheexample/ClientCacheExampleActivity.java]: Makes sense, I am not very familiar with the convention on android

**Pull request merged**